pallets/author-mapping/src/lib.rs

```diff
@@ -248,36 +253,39 @@ pub mod pallet {
 		#[pallet::weight(<T as Config>::WeightInfo::set_keys())]
 		pub fn set_keys(origin: OriginFor<T>, keys: (NimbusId, T::Keys)) -> DispatchResult {
 			let account_id = ensure_signed(origin)?;
-			let old_author_id =
-				Self::nimbus_id_of(&account_id).ok_or(Error::<T>::OldAuthorIdNotFound)?;
-			let stored_info = MappingWithDeposit::<T>::try_get(&old_author_id)
-				.map_err(|_| Error::<T>::AssociationNotFound)?;
-
-			ensure!(
-				account_id == stored_info.account,
-				Error::<T>::NotYourAssociation
-			);
-			let (new_author_id, new_keys) = keys;
+			let (author_id, keys) = keys;
 			ensure!(
-				MappingWithDeposit::<T>::get(&new_author_id).is_none(),
+				MappingWithDeposit::<T>::get(&author_id).is_none(),
```

Comment:
I thought the set_keys would work if already associated (being idempotent)

Reply:
This line checks if the new key has a mapping associated with it. That's why the error in next line is `AlreadyAssociated` because if the condition is false then the `author_id: NimbusId` has already been claimed by someone else.

Reply:
In that case, you should fail only if the `new_author_id != existing_author_id` right ?

Reply:
For naming convention, when there is a new and previous value, I would use a prefix for both:
`new_author_id`
`previous_author_id` or `existing_author_id` (whatever is more consistent with the rest of the code)

Reply:
> In that case, you should fail only if the new_author_id != existing_author_id right ?

Added this check TODO: link to where I added it

> For naming convention, when there is a new and previous value, I would use a prefix for both

I used `old_` prefix like the rest of the pallet for the existing/previous value